homeassistant/components/tractive/switch.py

```diff
@@ -0,0 +1,162 @@
+"""Support for Tractive switches."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from aiotractive.exceptions import TractiveError
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BUZZER,
+    ATTR_LED,
+    ATTR_LIVE_TRACKING,
+    CLIENT,
+    DOMAIN,
+    SERVER_UNAVAILABLE,
+    TRACKABLES,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+)
+from .entity import TractiveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class TractiveSwitchEntityDescription(SwitchEntityDescription):
+    """Class describing Tractive switch entities."""
+
+    method: str | None = None
+
+
+SWITCH_TYPES = (
+    TractiveSwitchEntityDescription(
+        key=ATTR_BUZZER,
+        name="Tracker Buzzer",
+        icon="mdi:volume-high",
+        method="async_set_buzzer",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LED,
+        name="Tracker LED",
+        icon="mdi:led-on",
+        method="async_set_led",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LIVE_TRACKING,
+        name="Live Tracking",
+        icon="mdi:map-marker-path",
+        method="async_set_live_tracking",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tractive switches."""
+    client = hass.data[DOMAIN][entry.entry_id][CLIENT]
+    trackables = hass.data[DOMAIN][entry.entry_id][TRACKABLES]
+
+    entities = [
+        TractiveSwitch(client.user_id, item, description)
+        for description in SWITCH_TYPES
+        for item in trackables
+    ]
+
+    async_add_entities(entities)
+
+
+class TractiveSwitch(TractiveEntity, SwitchEntity):
+    """Tractive switch."""
+
+    entity_description: TractiveSwitchEntityDescription
+
+    def __init__(self, user_id, trackables, description) -> None:
+        """Initialize switch entity."""
+        super().__init__(user_id, trackables.trackable, trackables.tracker_details)
```

Comment:
`trackables.trackable` is a bit difficult to read. Why not change `trackables` to `item`? You have used that before.